api/v1/infinispan_webhook_test.go

```diff
@@ -314,6 +314,42 @@ var _ = Describe("Infinispan Webhooks", func() {
 				"FieldValueForbidden", "spec.service.sites", "XSite not supported",
 			}}...)
 		})
+
+		It("Should convert XSite Host and Port spec fields to URL", func() {
```

Comment:
@ryanemerson where is this method invoked? I don't see any new test in Jenkins when I compare the results with `main`.

Reply:
These tests are executed via `make test`, the setup of the testsuite is handled by `webhook_suit_test.go` in the package. It's subtle, but you can see that the test coverage has increased with this PR.

PR:
```
./api/... -coverprofile cover.out
16:42:12  ok  	github.com/infinispan/infinispan-operator/api/v1	7.563s	coverage: 26.2% of statements
16:42:12  ok  	github.com/infinispan/infinispan-operator/api/v2alpha1	8.440s	coverage: 36.4% of statements
16:42:12  go test ./controllers/...
16:42:14  ok  	github.com/infinispan/infinispan-operator/controllers	0.041s
16:42:14  ?   	github.com/infinispan/infinispan-operator/controllers/constants	[no test files]
```

Main:
```
./api/... -coverprofile cover.out
22:16:30  ok  	github.com/infinispan/infinispan-operator/api/v1	7.802s	coverage: 25.4% of statements
22:16:30  ok  	github.com/infinispan/infinispan-operator/api/v2alpha1	8.465s	coverage: 36.4% of statements
22:16:30  go test ./controllers/...
22:16:32  ok  	github.com/infinispan/infinispan-operator/controllers	0.032s
22:16:32  ?   	github.com/infinispan/infinispan-operator/controllers/constants	[no test files]
```

I've added an additional commit to the PR setting the `-v` flag so that we can easily inspect how many steps are executed in the future

Reply:
I don't see any xsite reference in `webhook_test.go`. I'm wondering if it ever enters inside the `if` block.

Reply:
I think there might be some confusion here about the test I have added.

It's a Ginko Env Test that's executed as part of `make test`. These tests aren't executed against a "full" k8s cluster like kind, instead they're just executed against the loaded webhook/controllers in `webhook_suite_test.go` using the k8s api-server and etcd.

It has not relation to the E2E webhook tests in `test/e2e/webhook/webhook_test.go`. These tests are executed against a "full" k8s cluster, `kind` on the CI, using an OLM deployment.

The env tests are sufficient for testing webhook semantics, whereas the integration test is required to ensure that the webhook services etc work correctly when deployed via OLM.